RadioButtonGroup - MAUI Crash on Release InvalidCastException

### Description

Good evening,

MAUI Version : 6.0.300-rc.3.5667
VS Community 2022 - Version 17.3.0 Preview 1.0 (As Admin).

I spotted an issue on the application I'm trying to build.
An InvalidCastException is raised inside the constructor of RadioButtonGroupController.

It looks like the Element cast in the constructor is causing the issue ; for my own comprehesion I find it odd to have a cast to Element class on an ILayout interface ; does there is any explanation (regarding the Maui infrastructure) or is it a little mistake ?
I can't figure out why this is working in Debug mode.

This issue occurs only in Release mode (In VS 2022 -> drop down : Release).

Works when this tag is present in the csproj : <Optimize>False</Optimize>
Non-working if omitted or set to true.

This is my first BugReport if something is missing don't hesitate to tell me.
Repro attached to the ticket.

[MauiApp1.zip](https://github.com/dotnet/maui/files/8734634/MauiApp1.zip)

Kind Regards.

### Steps to Reproduce

1. Open the SLN
2. Check that you selected 'Release'
3. Clean Solution & remove the bin and obj folder
4. Press the green arrow (Start debugging) to start and deploy the project Pixel 2 - API28 (Android 9.0 - API28)
5. The app start and crash instantly in the emulator (I faced the same issue on a physical phone, samsung).
6. Press continue until you see the stack trace
-----------------------
7. Redo step 2 & 3
8. Remove the RadioButtonGroup.GroupName="priceType"  from both radiobutton (MainPage.xaml)
9. Do step 4
10. The app works

### Version with bug

Release Candidate 3 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 9.0 - API 28

### Did you find any workaround?

Removing the attached RadioButtonGroup and manage the exclusive state selection by databinding or code behind on the view.

### Relevant log output

```shell
[mono-rt] [ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidCastException: Specified cast is not valid.
[mono-rt]    at Microsoft.Maui.Controls.RadioButtonGroup.<>c.<.cctor>b__14_1(BindableObject b)
[mono-rt]    at Microsoft.Maui.Controls.BindableObject.GetValue(BindableProperty property)
[mono-rt]    at Microsoft.Maui.Controls.RadioButtonGroup.GetRadioButtonGroupController(BindableObject b)
[mono-rt]    at Microsoft.Maui.Controls.RadioButtonGroup.<>c.<.cctor>b__14_2(BindableObject b, Object o, Object n)
[mono-rt]    at Microsoft.Maui.Controls.BindableObject.SetValueActual(BindableProperty property, BindablePropertyContext context, Object value, Boolean currentlyApplying, SetValueFlags attributes, Boolean silent)
[mono-rt]    at Microsoft.Maui.Controls.BindableObject.SetValueCore(BindableProperty property, Object value, SetValueFlags attributes, SetValuePrivateFlags privateAttributes)
[mono-rt]    at Microsoft.Maui.Controls.BindableObject.SetValue(BindableProperty property, Object value, Boolean fromStyle, Boolean checkAccess)
[mono-rt]    at Microsoft.Maui.Controls.BindableObject.SetValue(BindableProperty property, Object value)
[mono-rt]    at MauiApp1.MainPage.InitializeComponent() in C:\Users\myUser\source\repos\MauiApp1\MauiApp1\Microsoft.Maui.Controls.SourceGen\Microsoft.Maui.Controls.SourceGen.CodeBehindGenerator\MainPage.xaml.sg.cs:line 33
[mono-rt]    at MauiApp1.MainPage..ctor(MainPageViewModel viewModel) in C:\Users\myUser\source\repos\MauiApp1\MauiApp1\MainPage.xaml.cs:line 11
[mono-rt]    at System.Reflection.RuntimeConstructorInfo.InternalInvoke(Object obj, Object[] parameters, Boolean wrapExceptions)
[mono-rt]    at System.Reflection.RuntimeConstructorInfo.DoInvoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
[mono-rt]    at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
[mono-rt]    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
[mono-rt]    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext, Microsoft.Extensions.DependencyInjection, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].VisitCallSiteMain(ServiceCallSite callSite, RuntimeResolverContext argument)
[mono-rt]    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
[mono-rt]    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext, Microsoft.Extensions.DependencyInjection, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].VisitCallSite(ServiceCallSite callSite, RuntimeResolverContext argument)
[mono-rt]    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
[mono-rt]    at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(Type serviceType)
[mono-rt]    at System.Collections.Concurrent.ConcurrentDictionary`2[[System.Type, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.Func`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope, Microsoft.Extensions.DependencyInjection, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].GetOrAdd(Type key, Func`2 valueFactory)
[mono-rt]    at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
[mono-rt]
```

--- a/src/Controls/src/Core/RadioButtonGroup.cs
+++ b/src/Controls/src/Core/RadioButtonGroup.cs
@@ -10,7 +10,7 @@ namespace Microsoft.Maui.Controls
 
 		static readonly BindableProperty RadioButtonGroupControllerProperty =
 			BindableProperty.CreateAttached("RadioButtonGroupController", typeof(RadioButtonGroupController), typeof(Maui.ILayout), default(RadioButtonGroupController),
-			defaultValueCreator: (b) => new RadioButtonGroupController((Maui.ILayout)b),
+			defaultValueCreator: (b) => new RadioButtonGroupController(b as Maui.ILayout),
 			propertyChanged: (b, o, n) => OnControllerChanged(b, (RadioButtonGroupController)o, (RadioButtonGroupController)n));
 
 		static RadioButtonGroupController GetRadioButtonGroupController(BindableObject b)

